domains/runtime/evm/src/lib.rs

```diff
@@ -911,6 +914,34 @@ fn construct_sudo_call_extrinsic(encoded_ext: Vec<u8>) -> <Block as BlockT>::Ext
     )
 }
 
+/// Returns `true` if this is a valid pallet-evm-tracker "contract creation allowed by" inherent
+/// call.
+fn is_valid_evm_contract_creation_allowed_by_call(
```

Comment:
we can always adapt the code when we have such requirements. No need to future proof this right now. This just adds more noise right now